src/DurableTask.Netherite/TransportLayer/EventHubs/EventHubsTransport.cs

```diff
@@ -289,39 +289,53 @@ public IEventProcessor CreateEventProcessor(PartitionContext context)
 
         async Task ITransportLayer.StopAsync()
         {
-            this.traceHelper.LogInformation("Shutting down EventHubsBackend");
-            this.shutdownSource.Cancel(); // initiates shutdown of client and of all partitions
+            this.traceHelper.LogInformation("EventHubsTransport is shutting down");
+            this.shutdownSource.Cancel(); // immediately initiates shutdown of client and of all partitions
 
-            if (this.hasWorkers)
-            {
-                this.traceHelper.LogDebug("Stopping partition and loadmonitor hosts");
-                await Task.WhenAll(
-                  this.StopPartitionHost(),
-                  this.loadMonitorHost.UnregisterEventProcessorAsync());
-            }
+            await Task.WhenAll(
+                 this.hasWorkers ? this.StopWorkersAsync() : Task.CompletedTask,
+                 this.StopClientsAndConnectionsAsync());
 
-            this.traceHelper.LogDebug("Stopping client process loop");
-            await this.clientProcessTask;
+            this.traceHelper.LogInformation("EventHubsTransport is shut down");
+        }
 
-            this.traceHelper.LogDebug("Stopping client");
-            await this.client.StopAsync();
+        async Task StopWorkersAsync()
+        {
+            this.traceHelper.LogDebug("EventHubsTransport is stopping partition and loadmonitor hosts");
+            await Task.WhenAll(
+              this.StopPartitionHostAsync(),
+              this.StopLoadMonitorHostAsync());
+        }
 
-            this.traceHelper.LogDebug("Closing connections");
+        async Task StopClientsAndConnectionsAsync()
+        {
+            this.traceHelper.LogDebug("EventHubsTransport is stopping client process loop");
+            await this.clientProcessTask;
+
+            this.traceHelper.LogDebug("EventHubsTransport is closing connections");
             await this.connections.StopAsync();
 
-            this.traceHelper.LogInformation("EventHubsBackend shutdown completed");
+            this.traceHelper.LogDebug("EventHubsTransport is stopping client");
+            await this.client.StopAsync();
         }
```

Comment:
do we want a log indicating the client _stopped_, not just that it began its stop operation?

Reply:
yes, that makes sense.